force-app/main/tools/classes/zChatterAgentTool.cls

```diff
@@ -0,0 +1,109 @@
+public class zChatterAgentTool implements IAgentTool {
+    public String getDescription() {
+        return 'Chatter Agent Tool for retrieving, inserting, updating, upserting, and deleting Chatter posts';
+    }
+
+    public Map<String, String> getParameters() {
+        return new Map<String, String>{
+            'operation' => 'The action to be performed: insert, delete, retrieve, update, or upsert on the Chatter post',
+            'Id' => 'The unique identifier of the Chatter post to be retrieved, updated, or deleted',
+            'parentId' => 'The unique identifier of the parent record, such as the Account record Id for a post on the Account record page',
+            'title' => 'The title of the feed item; for LinkPost type, this is the link name; can only be updated on posts of type QuestionPost',
+            'body' => 'The main text content of the Chatter post',
+            'isRichText' => 'Indicates whether the post body contains rich text formatting: true or false',
+            'linkUrl' => 'The web address for LinkPost type.',
+            'visibility' => 'Optional. AllUsers, InternalUsers. InternalUsers is the default.',
+            'type' => 'Optional. Default is TextPost. LinkPost, QuestionPost, or TextPost.',
+            'lastEditById' => 'The unique identifier of the Salesforce User who made the most recent edit. Optional',
+            'lastEditDate' => 'The date and time of the most recent edit, as a DateTime string. Optional.',
+            'revision' => 'The version number of the Chatter post as an Integer. Optional.',
+            'status' => 'The current state of the Chatter post: Published or PendingReview. The value Published  means publicly available while PendingReview means  awaiting approval.'
```

Comment:
This is a lot of context to add in for one tool!  It could impact the performance of the agent.  

If possible, it would be good to reduce the number of parameters and just set some reasonable defaults for them.

We could use the tool constructor to configure some defaults so that different agents might post to chatter in different ways.